contrib/net/http/http.go

```diff
@@ -57,13 +57,26 @@ func (mux *ServeMux) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 // WrapHandler wraps an http.Handler with tracing using the given service and resource.
 func WrapHandler(h http.Handler, service, resource string, opts ...Option) http.Handler {
+	opts = append(opts, WithStaticResourceNamer(resource))
+	return WrapHandlerWithDynamicResourceName(h, service, opts...)
+}
+
+// WrapHandlerWithDynamicResourceName wraps an http.Handler with tracing using the given service and generates a
+// resource name from the inbound request.  It probably makes more API sense for this to supplant WrapHandler but
+// for backcompat we leave that one, with its static naming, alone.  Unless maybe we were to allow resource to be nil?
+func WrapHandlerWithDynamicResourceName(h http.Handler, service string, opts ...Option) http.Handler {
```

Comment:
This change seems to be out of scope. Can you please undo it?

Reply:
This is a change to maintain existing behavior in the old code path and avoid confusion around the `resource` argument.

We didn't talk about it on the issue, so how should I resolve the differences in the argument list and/or preserve existing behavior?

Reply:
We can solve the problem through documentation. Simply adding something like:
```go
// [...]. If the WithResourceNamer option is provided as part of opts, it will take precedence over the resource argumeent.
```